modality-probe-capi/include/probe.h

```diff
@@ -73,6 +73,13 @@ typedef struct modality_probe_causal_snapshot {
     uint16_t reserved_1;
 } modality_probe_causal_snapshot;
 
+/*
+ * Function type for retrieving the next persistent sequence number
```

Comment:
Could we document more here about the expectations for the generated number? Also, the circumstances in which this function will be called?